src/control/control.cpp

```diff
@@ -42,16 +42,19 @@ ControlDoublePrivate::ControlDoublePrivate(ConfigKey key,
                        Stat::SAMPLE_VARIANCE | Stat::MIN | Stat::MAX),
           m_confirmRequired(false),
           m_pCreatorCO(pCreatorCO) {
-    initialize();
+    initialize(initialValue);
 }
 
-void ControlDoublePrivate::initialize() {
+void ControlDoublePrivate::initialize(double initialValue) {
     double value = 0;
     if (m_bPersistInConfiguration) {
         UserSettingsPointer pConfig = ControlDoublePrivate::s_pUserConfig;
-        if (pConfig != NULL) {
-            // Assume toDouble() returns 0 if conversion fails.
-            value = pConfig->getValueString(m_key).toDouble();
+        if (pConfig != nullptr) {
+            bool conversionWorked;
```

Comment:
ConfigObject now has a getValue method that does the type conversions so you can do:

```c++
value = defaultValue;
if (m_bPersistInConfiguration) { 
  UserSettingsPointer pConfig = ControlDoublePrivate::s_pUserConfig;
  if (pConfig != nullptr) { 
    value = m_pConfig->getValue(m_key, value);
  }
}
```